pkg/storage/metrics.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * Nuts crypto
+ * Copyright (C) 2020. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package storage
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/nuts-foundation/nuts-crypto/log"
+	errors2 "github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// in minutes
+const checkInterval = 15
+
+const label = "period"
+
+// ErrAlreadyStarted is returned when start is called on an already running monitor
+var ErrAlreadyStarted = errors.New("certificate monitor already started")
+
+// we register only one
+var metric *prometheus.GaugeVec
```

Comment:
ugh more hidden, shared state making unit testing hard(er). Can't we solve this by making Register() idempotent (by making it only register it when it's not already registered)

Reply:
Unfortunately you can't get a reference to a collector via the Prometheus Registry, you'll get an interface type. You could do some casting, but that's rather ugly as well.

Another challenge is that collectors are compared based on the name, not the supported labels. Only allowing the first to register would yield unexpected behaviour. Specifying which metrics is done at compile-time. So returning an error when this is done twice seems appropriate.